mp2-v1/tests/common/cases/query.rs

```diff
@@ -161,35 +148,43 @@ async fn test_query_mapping(
     let mut exec_query = parsil::executor::generate_query_execution(&mut parsed, &settings)?;
     let query_params = exec_query.convert_placeholders(&query_info.placeholders);
     let res = table
-        .execute_row_query(&exec_query.apply().to_string(), &query_params)
+        .execute_row_query(
+            &exec_query.normalize_placeholder_names().to_pgsql_string(),
+            &query_params,
+        )
         .await?;
     info!(
         "Found {} results from query {}",
         res.len(),
-        exec_query.query.to_string()
+        exec_query.query.to_display()
     );
     print_vec_sql_rows(&res, SqlType::Numeric);
 
     let pis = parsil::assembler::assemble_dynamic(&parsed, &settings, &query_info.placeholders)
         .context("while assembling PIs")?;
 
-    let rows_query = parsil::keys_in_index_boundaries(&query_info.query, &settings, &pis.bounds)
-        .context("while genrating keys in index bounds")?;
-    println!(" -- touched rows query: {:?}", rows_query.query.to_string());
-    let initial_ph = rows_query.convert_placeholders(&query_info.placeholders);
-    println!("initial_ph: {:?}", initial_ph);
+    let mut rows_query =
+        parsil::keys_in_index_boundaries(&query_info.query, &settings, &pis.bounds)
+            .context("while genrating keys in index bounds")?;
+    println!(
+        " -- touched rows query: {:?}",
+        rows_query.query.to_display()
+    );
     let big_row_cache = table
         .row
         .wide_lineage_between(
-            &rows_query.query.to_string(),
+            &rows_query
+                .interpolate(&settings, &query_info.placeholders)?
+                .to_pgsql_string(),
             (query_info.min_block as Epoch, query_info.max_block as Epoch),
         )
         .await?;
     // We set the epoch at which we request all the lineages - that's a fixed epoch
     // and we set the generate_series according to the query
     let current_epoch = table.index.current_epoch();
+    // Integer default to i32 in PgSQL, they must be cast to i64, a.k.a. BIGINT.
     let index_query = format!(
-        "SELECT {current_epoch} as epoch, generate_series({}, {}) AS key",
+        "SELECT {current_epoch}::BIGINT as {EPOCH}, generate_series({}::BIGINT, {}::BIGINT) AS {KEY}",
```

Comment:
Any chance we can expose that from parsil ? 
so we'd get something like
* `parsil::query_for_row_tree`
* `parsil::query_for_index_tree` ?